### DISCUSSION TITLE
Question about RxEngineType

### DISCUSSION THREAD

**Discussion by zion830** [euphony-io/euphony#202] [Q&A] [answered]
Hi! I have a question about the following code
```java
    public void setRxEngineType(RxEngineType type) {
        this.rxEngineType = type;
    }
```
```java
    public boolean listen() {
        if (rxEngineType == RxEngineType.EUPHONY_JAVA_ENGINE)
            return listenOnJava();
        else
            return listenOnNative();
    }
```
* What's the difference of `listenOnJava` and `listenOnNative`?
* The default is `EUPHONY_JAVA_ENGINE`, then when should I change this to `EUPHONY_NATIVE_ENGINE`?

**Comment by designe** [comment, marked as answer]
Hi @zion830!
Sorry for late reply. I've been very busy this week, so I'm just posting an answer.

#### What's the difference of `listenOnJava` and `listenOnNative`?
- `listenOnJava` & `EUPHONY_JAVA_ENGINE`
https://github.com/euphony-io/euphony/blob/58c7b827c5bc76fad1390559d3abfe6a0a7bf93d/euphony/src/main/java/co/euphony/rx/EuRxManager.java#L64
Actually, `listenOnJava` function is the core of rx feature. it is the start point of rx.
This is based on java. From `recording audio` to `data encoding`, all feature are made by java without FFT implementation.

- `listenOnNative` & `EUPHONY_NATIVE_ENGINE`
https://github.com/euphony-io/euphony/blob/58c7b827c5bc76fad1390559d3abfe6a0a7bf93d/euphony/src/main/java/co/euphony/rx/EuRxManager.java#L92
`listenOnNative` only has a shell. Currently, rx nativeization is in progress, when it is completed and stabilized to some extent, it will replace the `listenOnJava` function completely. But now, you can ignore it now :)

Please note that `listenOnJava` is currently used by `default`.

**Reply by zion830** [reply]
I understand! thanks for your answer! :D
